api/src/main/java/org/apache/iceberg/expressions/ExpressionVisitors.java

```diff
@@ -59,6 +59,10 @@ public R or(R leftResult, R rightResult) {
   }
 
   public abstract static class BoundExpressionVisitor<R> extends ExpressionVisitor<R> {
+    protected <T> Literal<T> toLiteral(T val) {
+      return Literals.from(val);
```

Comment:
I don't see value in adding this method. Callers can use `Literals.from` directly, but the visitors should not even be converting values to literals.

Reply:
Removed.